### ISSUE TITLE
Training ST-GCN for evaluation

### ISSUE THREAD

**Issue by mkstmyk** [soomean/Diverse-Motion-Stylization#2]
Thanks for the great work.
If I'm not mistaken, you've trained ST-GCN to compute FMD and recognition accuracy as written in the paper.
Do you have any plans to upload ST-GCN model trained on the datasets used in the paper?
As the data format is different from the on used in the original ST-GCN, I really appreciate if you can share the code for importing the data to ST-GCN. At least, any tips for reproducing it would be very helpful.
Thank you for your consideration.

**Comment by soomean** [comment]
Hi, @mkstmyk
Sorry, it may take a while to clean up the code for evaluation.
But meantime you can look up the original ST-GCN code that I mainly referenced:
https://github.com/yysijie/st-gcn/blob/221c0e152054b8da593774c0d483e59befdb9061/net/st_gcn.py#L9-L116

I assume that the evaluation requires only minor modifications on it.

**Comment by mkstmyk** [comment]
Hi, @soomean 
Thank you for your quick reply! I'll have a look at the original ST-GCN code.
Just to be sure, please allow me to ask you a few more questions for data processing:
- If I'm not mistaken, you've trained ST-GCN on the [datasets](https://drive.google.com/drive/folders/1C-_iZJj-PSUWZwh25yAsQe1tLpPm9EZ5?usp=sharing) provided by Aberman, right?
- And you also preprocessed them by following Holden and stored them as preprocessed_styletransfer_classify/generate.npz to train your model. Is this correct?
- In the training of ST-GCN for the evaluation, how did you give action classes to the model because only style labels are provided in Aberman's dataset?
- Is the preprocessing is not needed for ST-GCN training?
- In my understanding, ST-GCN needs trajectories of joints while BVH only provides angles. Can this gap be filled by Animation.transforms_global?
- As the data size is not so huge, I'm wondering whether you trained ST-GCN from the scratch or fine-tuned to the dataset.

Sorry for asking many questions. I wish I could attend the conference and ask those there.
Thank you again in advance.

**Comment by soomean** [comment]
- The model is trained on the dataset provided by Xia et al.

**Comment by soomean** [comment]
- The model is trained on the dataset provided by Xia et al. (You can request the original authors for it, but looks like it's already included in the dataset you linked to). I used Aberman's to test transferring style to long-term heterogeneous motions.
- Xia dataset has action labels, so it was able to train a recognition model according to that label.
- The preprocessing is required for training ST-GCN.
- The position is included in the preprocessed data format. We just used that information for ST-GCN. (You can definitely use the Animation library to get the data representation you want.)
- I trained ST-GCN from scratch since the dataset size is not huge.

Hope my answer would help!

**Comment by mkstmyk** [comment]
Thank you very much for your helpful answers. Let me confirm about the way you preprocessed the data for ST-GCN. Is it what written in the paragraph "Preprocessing" in the chapter 4 in the paper and implemented in the later half of preprocess function in export_dataset.py?
Sorry for bothering you.

**Comment by soomean** [comment]
Yes, it is. Please refer to the paper for more details.
Thanks for having an interest in our work.

**Comment by mkstmyk** [comment]
Thank you for replying again and again! Let me close this at this point.

**Comment by mkstmyk** [comment]
Thank you for replying again and again! Let me close this at this point.

**Comment by LaLaLailalai** [comment]
> Yes, it is. Please refer to the paper for more details. Thanks for having an interest in our work.

Hi @soomean 

As far as I understand, the feature is the output after the last pooling layer of ST-GCN？
